homeassistant/components/statistics/sensor.py

```diff
@@ -162,70 +169,81 @@ def valid_binary_characteristic_configuration(config):
 )
 
 
-async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    async_add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
     """Set up the Statistics sensor."""
 
     await async_setup_reload_service(hass, DOMAIN, PLATFORMS)
 
     async_add_entities(
-        [
+        new_entities=[
             StatisticsSensor(
-                source_entity_id=config.get(CONF_ENTITY_ID),
-                name=config.get(CONF_NAME),
+                source_entity_id=str(config.get(CONF_ENTITY_ID)),
```

Comment:
This can't be right, `dict.get` may return `None` and we don't want to stringify that.

In this case, `CONF_ENTITY_ID` is not optional though, so we should do `[]`, not `.get()`

Same comment for the other constructor parameters.

Reply:
Interesting. I had the same base thought. Every required parameter or optional parameter with a default should exist and be a string, at minimum "". In the case of unique_id I did not add any conversion because it could indeed be `None`.

I don't get your point with `[]`. The keys exist and the return of `None` isn't solved by it...

Reply:
Had a closer look. Currently the only parameter that can be `None` is `config.get(CONF_MAX_AGE)`. All the others are constrained in a certain way and are an empty string in the worst case.

Of course I agree, this could change at any point and we shouldn't convert None to "None". What now?

Btw. regarding the empty strings: Check this horrible valid testcase. That's shouldn't be allowed, correct?
 https://github.com/home-assistant/core/pull/60997/commits/f908b71de06ce4d5d1dd76a88162e10a78d3db9e

Reply:
> I don't get your point with []. The keys exist and the return of None isn't solved by it

By doing `[]` (or` __getitem__` really) we assert that the item exists, and if it doesn't, it throws.
We've already validated `config[CONF_ENTITY_ID]` - and all the other configuration variables - so there's no need to stringify.